src/Utils.ts

```diff
@@ -59,59 +120,102 @@ export function isSecondEventLater(a: TypedEvent, b: TypedEvent): boolean {
 
 export function isRegistrationValid(
   relayData: IRelayHub.RelayManagerDataStruct | undefined,
-  config: AppConfig,
   managerAddress: string
 ): boolean {
-  const portIncluded: boolean = config.url.indexOf(':') > 0;
+  const appConfig = config.get<AppConfig>('app');
+
+  const portIncluded: boolean = appConfig.url.indexOf(':') > 0;
 
   if (relayData) {
     const manager = relayData.manager as string;
 
     return (
       isSameAddress(manager, managerAddress) &&
       relayData.url.toString() ===
-        config.url.toString() +
-          (!portIncluded && config.port > 0 ? ':' + config.port.toString() : '')
+        appConfig.url.toString() +
+          (!portIncluded && appConfig.port > 0
+            ? ':' + appConfig.port.toString()
+            : '')
     );
   }
 
   return false;
 }
 
-/**
- * @returns maximum possible gas consumption by this relay call
- * Note that not using the linear fit would result in an Inadequate amount of gas
- * You can add another kind of estimation (a hardcoded value for example) in that "else" statement
- * if you don't then use this function with usingLinearFit = true
- */
-export function estimateMaxPossibleRelayCallWithLinearFit(
-  relayCallGasLimit: BigNumberish,
-  tokenPaymentGas: BigNumberish,
-  addCushion = false
-): BigNumber {
-  const cushion = addCushion ? ESTIMATED_GAS_CORRECTION_FACTOR : 1.0;
-
-  const bigRelay = BigNumberJs(relayCallGasLimit.toString());
-  const bigTokenPayment = BigNumberJs(tokenPaymentGas.toString());
-
-  let estimatedCost: BigNumberJs;
-
-  if (bigTokenPayment.isZero()) {
-    // Subsidized case
-    // y = a0 + a1 * x = 85090.977 + 1.067 * x
-    const a0 = BigNumberJs('85090.977');
-    const a1 = BigNumberJs('1.067');
-    estimatedCost = a1.multipliedBy(bigRelay).plus(a0);
-  } else {
-    // y = a0 + a1 * x = 72530.9611 + 1.1114 * x
-    const a0 = BigNumberJs('72530.9611');
-    const a1 = BigNumberJs('1.1114');
-    estimatedCost = a1.multipliedBy(bigRelay.plus(bigTokenPayment)).plus(a0);
-  }
+export async function isContractDeployed(address: string): Promise<boolean> {
+  const provider = getProvider();
+
+  const code = await provider.getCode(address);
 
-  const costWithCushion = estimatedCost
-    .multipliedBy(cushion.toString())
-    .decimalPlaces(0, BigNumberJs.ROUND_CEIL);
+  // Check added for RSKJ: when the contract does not exist in RSKJ it replies to the getCode call with 0x00
+  return code !== '0x' && code !== '0x00';
+}
+
+export function getServerConfig(): ServerConfigParams {
+  const contractsConfig: ContractsConfig = config.get('contracts');
+  const appConfig: AppConfig = config.get('app');
+  const blockchainConfig: BlockchainConfig = config.get('blockchain');
```

Comment:
Same as for other parts, I'd use constants to store config keys

Reply:
updated